lib/smack/SmackInstGenerator.cpp

```diff
@@ -145,39 +145,79 @@ void SmackInstGenerator::processInstruction(llvm::Instruction& inst) {
 }
 
 void SmackInstGenerator::processIntrinsicCall(llvm::IntrinsicInst* ii) {
+  #define OVER_APPROXIMATE \
```

Comment:
There must be a way to avoid preprocessor macros here…

Reply:
I added more macros :)

Why would we want to avoid them?

Reply:
Search the web for “macros are evil” — there are several well-known reasons, and I don’t think I should have to explain them.

If anything, I would implore you to justify why you should use macros here and not functions. Sometimes there are valid reasons for doing so.

Reply:
I removed all the macros defined in this function.